### PR TITLE
Add label filtering to v1alpha3 branch

### PR DESCRIPTION
**Pull request by MarcelMue**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: api/v1alpha3/common_types.go [hunk 1]
```diff
@@ -38,6 +38,12 @@ const (
 	// on the reconciled object.
 	PausedAnnotation = "cluster.x-k8s.io/paused"
 
+	// WatchLabel is a label othat can be applied to any Cluster API object.
```
Comment (anvddriesch) [inline api/v1alpha3/common_types.go]:
```suggestion
	// WatchLabel is a label that can be applied to any Cluster API object.
```

**Review by anvddriesch — APPROVED** [review]
If this is how this is done upstream it seems good.

File: Makefile [hunk 2]
```diff
@@ -69,21 +69,21 @@ CLOUDINIT_GENERATED := $(CLOUDINIT_PKG_DIR)/zz_generated.bindata.go
 CLOUDINIT_SCRIPT := $(CLOUDINIT_PKG_DIR)/kubeadm-bootstrap-script.sh
 
 # Define Docker related variables. Releases should modify and double check these vars.
-REGISTRY ?= gcr.io/$(shell gcloud config get-value project)
+REGISTRY ?= quay.io
```
Comment (kubasobon) [inline Makefile:72]: dockerhub.io ?

Comment (MarcelMue) [reply Makefile:72]: Later - I don't know how fast the syncing is

Comment (MarcelMue) [reply Makefile:72]: Or do you know how we can push to dockerhub directly?

**Review by tfussell — APPROVED** [review]

**Comment by anvddriesch** [comment]
Can I pull the trigger on this and merge?

**Comment by MarcelMue** [comment]
> Can I pull the trigger on this and merge?

You can pull whichever trigger you want - be aware that this merges into branch `release-0.3` though!

**Comment by anvddriesch** [comment]
Is our forks master on v1alpha3? In that case I'll change it to master.

**Comment by MarcelMue** [comment]
> Is our forks master on v1alpha3? In that case I'll change it to master.

I don't think that would be good - the master of this fork is specific to Azure before CAPI. Because in Azure we used v1alpha3 - only for imports basically.

**Pull request merged**